src/include/ipc/stream.h

```diff
@@ -95,9 +95,11 @@ struct sof_ipc_stream_params {
 	uint32_t host_period_bytes;
 	uint16_t no_stream_position; /**< 1 means don't send stream position */
 	uint8_t cont_update_posn; /**< 1 means continuous update stream position */
+	uint16_t ext_data_length; /**< 0 means no extended data */
```

Comment:
this structure is packed, please, move this up before `cont_update_posn` to avoid unaligned access especially on ARM

Reply:
@lyakh you have a good point. but i don't like changing the order of fields as it is already part of the ABI. I have added 1 uint8_t reseved field. See the new updated code.

Reply:
@dbaluta ah, right sure, but then let's put this 16-bit field after the reserved bytes to avoid splitting them. Even though you're unlikely to use 3 bytes together, still it might happen that exactly 3 bytes would be needed for an array or a structure